### PR TITLE
Fix sentry sourcemaps

### LINKED ISSUE #19289: Fix sentry source maps: part 1
Sentry errors are currently minified code.

To improve developer experience during sentry error analysis and debugging, we want to ensure source maps are correctly recognized by the sentry pipeline.

The exact cause is unknown.

Sentry CLI has been recently updated. However, that may not be enough. We should inspect the sentry upload script.

Actions to take as part of this ticket:
- upload source maps to personal instance, see if that fixes problem
- if that doesn't work, update sentry cli and upload again
- if that doesn't work, read latest documentation and verify sentry sourcemap upload script matches documentation

### PR DESCRIPTION
**Pull request by danjm**
Fixes https://github.com/MetaMask/metamask-extension/issues/19289

## Explanation

We are seeing errors like this in sentry:

Screenshot from 2023-07-21 11-04-32

The fix is to add a `/` before the file path in the sentry event report, and before the url-prefix in the sourcemap upload. While this was not necessary at some point in the past, and the change to this being a requirement is seemingly not documented, adding a trailing slash is sufficient for the filepath to be read by sentry as a valid url.

The documentation does describe how the `url-prefix` flag in the cli defaults to `~/` as the url prefix. However, it does not make clear that if setting a custom prefix, it needs to be a valid file path and that neither `~/` or `/` are prefixed to the custom prefix passed to the cli flag. In our case, either `--url-prefix '~/metamask'` or `--url-prefix '/metamask'` would work, but the former violates our `lint:shellcheck` rules so we go with the latter.

It also seems that each error report frame's `filename` property needs to align with the file path that is uploaded by the cli. Alternatively, we could separately set an `abs_path` property on each error report frame, but the solution in this PR is the minimum viable change.

I pieced this solution together by reading different sentry bug reports and by some trial and error using the manual testing process describe below. Some of the sentry bug reports that were helpful to me were:
- https://github.com/robertcepa/toucan-js/issues/54#issuecomment-1055955708
- https://github.com/getsentry/sentry-cli/issues/894#issuecomment-768600143
- https://github.com/getsentry/sentry-webpack-plugin/issues/62#issuecomment-423746322 and https://github.com/getsentry/sentry-webpack-plugin/issues/62#issuecomment-423788005

Note that the scuttling related changes currently in this PR are temporary, as we want to merge them to develop first as part of https://github.com/MetaMask/metamask-extension/pull/20090. They are needed to successfully manually test this PR

Also note that this PR updates the sentry cli version. I am not certain that is required to fix the sourcemaps problem, however it was recommended as a first step in debugging sourcemap problems, and all my testing was done with the updated cli.

### Before

Screenshot from 2023-07-21 14-57-10

### After

Screenshot from 2023-07-21 14-55-24

## Manual Testing Steps

The below test steps should produce the "Before" error on develop and should show a successfully source mapped stack trace on this branch.

1. Checkout the desired branch (`develop` or this one)
2. Edit some code in the `app/` directory so that you can reliably have an error thrown from the background
3. Follow the instructions for "Debugging in Sentry" found here https://github.com/MetaMask/metamask-extension/tree/develop/development#debugging-in-sentry
4. When onboarding to the metamask install, be sure to agree to `Participate In MetaMetrics`
5. Take the action to have an error thrown
6. Observe the sentry instance you created in step 3

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: development/build/index.js [hunk 1]
```diff
@@ -99,6 +99,8 @@ async function defineAndRunBuildTasks() {
       'navigator',
       'harden',
       'console',
```
Comment (danjm) [inline development/build/index.js:101]: We can remove the commit that changes this file once https://github.com/MetaMask/metamask-extension/pull/20090 is merged

**Comment by danjm** [comment]
Blocked by https://github.com/MetaMask/metamask-extension/pull/20090

**Comment by Gudahtt** [comment]
>In our case, either `--url-prefix '~/metamask'` or `--url-prefix '/metamask'` would work, but the former violates our `lint:shellcheck` rules so we go with the latter.

ShellCheck probably thinks `~` is meant to expand to the home directory, but the single quotes are preventing this expansion. Note that we can disable ShellCheck warnings in cases like this, it supports ignore comments.

Absolute path seems OK too though

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by darkwing — APPROVED** [review]

**Pull request merged**